packages/dataviews/src/dataform-controls/boolean.tsx

```diff
@@ -17,7 +20,21 @@ export default function Boolean< Item >( {
 	const { id, getValue, label } = field;
 
 	return (
-		<ToggleControl
+		<ValidatedToggleControl
+			required={ !! field.isValid.required }
```

Comment:
The way the `ValidatedToggleControl` works is this: when it is required, the only valid value is `true`.